dev/TEAL.md

```diff
@@ -189,6 +195,14 @@ _available_.
  * Since v7, the account associated with any contract present in the
    `txn.ForeignApplications` field is _available_.
 
+ * A Box is _available_ to an Approval Program if _any_ transaction in
+   the same group contains a box reference that denotes the box. The
+   index `i` in the box reference refers to the `ith` application in
```

Comment:
How about:

 * A Box is _available_ to an Approval Program if _any_ transaction in
   the same group contains a box reference (`txn.Boxes`) that denotes
   the box. A box reference contains an index `i`, and name `n`. The
   index refers to the `ith` application in the transaction's
   ForeignApplications array, with the usual convention that 0
   indicates the application ID of the app called by that
   transaction. No box is ever _available_ to a ClearStateProgram.

Reply:
Could you explain how this will work for multiple ITXN calls ?
Outer Txn 0 which calls app A, app A calls B, B calls C and C uses box storage.
What has to be passed in to each of the app calls?  The outer txn foreign arrays, through to C.
Other references allow index based references or address/id references and this is critical IMO.
Boxes breaking this and only allowing indices is breaking this pattern and concerns me greatly if true.
Requiring indexes to be carried through multiple inner transactions with those inner transactions playing no part in the construction of their message, yet them having to specify an index something else defined - will be a nightmare IMO.  Assuming I'm understanding the issue here.  

Simply - a caller of A should not have to define a magic reference in an exact foreign 'slot' because A calls B which then called C and C expects its box application reference to be in index 1 - yet A or B might rely on that.

Reply:
Looks good to me

Reply:
>  will be a nightmare IMO. Assuming I'm understanding the issue here.

I hope that I can help you sleep.  It is the opposite of your fears.  There are _no_ uses of the indexes for box refs in AVM bytecode. Boxes are always used by name.  In your example, if app C is going to use a box named "hello", then the top level txn would contain `{ ForeignApps: [ B, C ], Boxes: [ (2, "hello") ] }`

This indicates that the 2nd app id (C) will have access to the box in C named "hello".  Nothing needs to be done when calling B or when calling C to make that happen. C gets access because the appropriate box ref was created at the top-level.

Reply:
Ok, so the inner transaction from B to C won't specify any foreign params at all?  Got it.
This seems fine then - I thought the inner transactions would have to specify these args as well.
Panic has been dispelled.  👍